test/functional/bulk.test.js

```diff
@@ -2003,4 +2003,103 @@ describe('Bulk', function () {
         );
     });
   });
+
+  describe('Bulk operation transaction rollback', () => {
+    /** @type {import('../../src/index').MongoClient} */
+    let client;
+    /** @type {import('../../src/index').Collection<{ answer: number }>} */
+    let collection;
+
+    beforeEach(async function () {
+      const config = this.configuration;
+      client = config.newClient();
+      await client.connect();
+
+      try {
+        await client
+          .db('bulk_operation_writes_test')
+          .collection('bulk_write_transaction_test')
+          .drop();
+      } catch (_) {
+        // do not care
+      }
+
+      collection = await client
+        .db('bulk_operation_writes_test')
+        .createCollection('bulk_write_transaction_test');
+
+      await collection.deleteMany({});
+    });
+
+    it('should abort ordered/unordered bulk operation writes', {
+      metadata: { requires: { mongodb: '>= 4.2', topology: ['replicaset'] } },
+      async test() {
+        const session = client.startSession();
+        session.startTransaction({
+          readConcern: { level: 'local' },
+          writeConcern: { w: 'majority' }
+        });
+
+        let bulk = undefined;
+
+        // 1.
+        bulk = collection.initializeUnorderedBulkOp({ session });
+        bulk.insert({ answer: 42 });
+        await bulk.execute();
+
+        // 2.
+        bulk = collection.initializeOrderedBulkOp({ session });
+        bulk.insert({ answer: 43 });
+        await bulk.execute();
+
+        await session.abortTransaction();
+        await session.endSession();
+
+        const documents = await collection.find().toArray();
+
+        expect(documents).to.be.lengthOf(
+          0,
+          'bulk operation writes were made outside of transaction'
+        );
+
+        await client.close(); // session leak checker doesn't let you put the close in an after hook
```

Comment:
I'm pretty sure I've seen client close used in an afterEach, and it really does belong there; otherwise the cleanup is not guaranteed - in fact it would fail if that expect fails

Reply:
Yea I'm not sure what error I was seeing that made me think this, its gone now